### PR TITLE
Fix an issue in cudf::row_bit_count involving structs and lists at multiple levels.

### PR DESCRIPTION
**Pull request by nvdbaranec**
`row_bit_count` keeps track of a stack of "branches" which represent a span of rows to be included in the computed size.  As you traverse through a hierarchy of lists, that span of rows is maintained as a stack.  The code that was handling jumping out from the bottom of a stack to a new column was making the faulty assumption that the jump was only 1 level up.

### CODE REVIEW

**Review by ttnghia — APPROVED** [review]

**Comment by ttnghia** [comment]
Do you have any link to any open issue?

**Review by hyperbolic2346 — APPROVED** [review]

File: cpp/src/transform/row_bit_count.cu [hunk 1]
```diff
@@ -409,7 +409,8 @@ __global__ void compute_row_sizes(device_span<column_device_view const> cols,
   auto const num_rows = output.size();
   if (tid >= num_rows) { return; }
 
-  // branch stack. points to the last list prior to branching.
+  // branch stack. points to the last span prior to branching. a branch occurs only
+  // when we are inside of a list contained within a struct column.
```
Comment (davidwendt) [inline cpp/src/transform/row_bit_count.cu:413]:
```suggestion
  // my_branch_stack points to the last span prior to branching. A branch occurs only
  // when we are inside of a list contained within a struct column.
```

**Review by abellina — APPROVED** [review]
LGTM

**Comment by ttnghia** [comment]
Rerun tests.

**Comment by nvdbaranec** [comment]
rerun tests

**Comment by nvdbaranec** [comment]
@gpucibot merge

**Pull request merged**